src/resource_api/schema.py

```diff
@@ -190,8 +190,21 @@ def get_schema(self):
 
 
 class IndexableField(BaseSimpleField):
+    """
+    Base class for most of the primitive fields.
+
+    choices (list|tuple = None)
+        Predefined list of possible values for the field. ValidationError will be raised if field value doesn't
+        match any of the defined values.
+    choice_labels (list|tuple = None)
```

Comment:
I see your point and actually I had thought about it myself. My concern regarding the dict is the correct ordering of the choices: how can we guarantee correct ordering of the items? For some use cases like UI automation the correct ordering of choices could be crucial.

Order of items in python dict seems to be implementation specific. I see two options to address this issue:
   1. choices with labels defined as list of key-value pair tuples
   2. choices defined as enums

In both cases choices and labels would be split to separate lists in schema output, in order to preserve correct ordering and especially to maintain backward compatibility.

Somehow I dislike the idea of interpreting the contents of the choice list to guess whether it should include labels or not. Enum would support ordering and also deal with duplicate values.